lib/checkr/report.rb

```diff
@@ -11,6 +11,8 @@ class Report < APIResource
     attribute :candidate, :Candidate
     attribute_writer_alias :candidate_id, :candidate
 
+    attribute :adverse_items, APIList.constructor(:AdverseItem)
```

Comment:
attribute :adverse_items, :AdverseItemList, :nested => true, :default => {}
attribute_writer_alias :adverse_item_ids, :adverse_items

Reply:
APIList works in a case when the parent object has id/ids of the child attribute. Our public API doesn't return adverse_item_ids as part of the report object, so in this case there is a need to create an `AdverseItemList` (same idea as behind Candidate owning Documents and being able to access them through `DocumentList`). So you'll need to create an additional file `adverse_item_list.rb` with ```module Checkr
  class AdverseItemList < APIList

    attribute :next_href
    attribute :previous_href
    attribute :count
    attr_accessor :parent

    api_instance_method :all, :get

    def self.construct(json, parent=nil)
      lambda = constructor(:AdverseItem)
      instance = lambda.call(json)
      instance.parent = parent if parent
      instance.clear_changed_attributes
      instance
    end

    def path
      parent.path + "/adverse_items"
    end

  end
end```

Reply:
Thanks for clarifying, I wasn't sure about the difference between the two.